resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -1,115 +1,407 @@
-                "ec2:GetTransitGatewayRouteTableAssociations",
-                "ec2:GetTransitGatewayRouteTablePropagations",
-                "ec2:ModifyInstanceAttribute",
-                "ec2:SearchLocalGatewayRoutes",
-                "ec2:SearchTransitGatewayMulticastGroups",
-                "ec2:SearchTransitGatewayRoutes",
-                "ec2:StartNetworkInsightsAnalysis",
-                "elasticloadbalancing:ConfigureHealthCheck",
-                "kms:CreateGrant",
-                "s3:GetBucketTagging",
-                "s3:GetObjectAcl",
-                "s3:GetObjectTagging",
-                "s3:ListAllMyBuckets",
-                "tiros:CreateQuery",
-                "tiros:GetQueryAnswer",
-                "tiros:GetQueryExplanation",
-                "ec2:RunInstances",
-                "ec2:DescribeInstanceTypes",
-                "ec2:CreateTags",
-                "ec2:GetConsoleOutput"
-            ],
-            "Resource": "*"
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "DescribeSnapShot",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeSnapshots",
+        "ec2:DescribeSnapshotAttribute"
+      ],
+      "Resource": "arn:aws:ec2:*:*:snapshot/*",
+      "Condition": {
+        "StringEquals": {
+          "aws:ResourceTag/red-hat-managed": "true"
+        }
+      }
+    },
+    {
+      "Sid": "CreateSnapShotsAllInstances",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CreateSnapshots"
+      ],
+      "Resource": "arn:aws:ec2:*:*:instance/*",
+      "Condition": {
+        "StringEquals": {
+          "aws:ResourceTag/red-hat-managed": "true"
+        }
+      }
+    },
+    {
+      "Sid": "ManageSnapshot",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CopySnapshot",
+        "ec2:CreateSnapshot"
+      ],
+      "Resource": "arn:aws:ec2:*:*:snapshot/*",
+      "Condition": {
+        "StringEquals": {
+          "aws:RequestTag/red-hat-managed": "true"
+        }
+      }
+    },
+    {
+      "Sid": "SendSerialConsoleSSHPublicKey",
+      "Effect": "Allow",
+      "Action": [
+        "ec2-instance-connect:SendSerialConsoleSSHPublicKey",
+        "ec2:GetSerialConsoleAccessStatus"
+      ],
+      "Resource": "arn:aws:ec2:*:*:instance/*",
+      "Condition": {
+        "StringEquals": {
+          "aws:ResourceTag/red-hat-managed": "true"
+        }
+      }
+    },
+    {
+      "Sid": "DescribeNetworkInsightsPath",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeNetworkInsightsPaths"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "DeleteNetworkInsightsPath",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DeleteNetworkInsightsPath",
+        "ec2:StartNetworkInsightsAnalysis"
+      ],
+      "Resource": "arn:aws:ec2:*:*:network-insights-path/*",
+      "Condition": {
+        "StringEquals": {
+          "aws:RequestTag/red-hat-managed": "true"
```

Comment:
These all have this tag?

Reply:
I have removed the tag for '"SendSerialConsoleSSHPublicKey", for actions creating "arn:aws:ec2:*:*:snapshot/*","arn:aws:ec2:*:*:network-insights-path/*" I have added then to "CreateTags" so they have the tag upon creation, the other actions can use the tag.
`
"Sid": "CreateRedHatManagedTag",
264       "Effect": "Allow",
265       "Action": [
266         "ec2:CreateTags"
267       ],
268       "Resource": [
269         "arn:aws:ec2:*:*:instance/*",
270         "arn:aws:ec2:*:*:snapshot/*",
271         "arn:aws:ec2:*:*:network-insights-path/*"
272       ],
273       "Condition": {
274         "StringEquals": {
275           "ec2:CreateAction": [
276               "RunInstances",
277               "CreateSnapshots",
278               "CreateSnapshot",
279               "CreateNetworkInsightsPath"
280           ],
281           "aws:RequestTag/red-hat-managed": "true"
282         },
283         "ForAllValues:StringEquals": {
284           "aws:TagKeys": "red-hat-managed"
285         }
286       }

`

Reply:
Why did you place CreateNetworkInsightsPath into that Statment? There's already another statement for it on line 85

Reply:
I think this condition for  "ec2:CreateTags"  allows  CreateNetworkInsightsPath to be able to tag the resource  "red-hat-managed":"true", no? https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/supported-iam-actions-tagging.html

Reply:
Are you calling CreateTag as a separate API call? Or is the tag included in the CreateNetworkInsightsPath call?

Reply:
Yeah i think it will